src/liballoc/collections/vec_deque.rs

```diff
@@ -57,11 +57,74 @@ pub struct VecDeque<T> {
     buf: RawVec<T>,
 }
 
+struct PairSlices<'a, 'b, T> {
```

Comment:
```suggestion
/// PairSlices pairs up equal length slice parts of two deques
///
/// For example, given deques "A" and "B" with the following division into slices:
///
/// A: [0 1 2] [3 4 5]
/// B: [a b] [c d e]
///
/// It produces the following sequence of matching slices:
///
/// ([0 1], [a b])
/// ([2], [c])
/// ([3 4], [d e])
///
/// and the uneven remainder of either A or B is skipped.
struct PairSlices<'a, 'b, T> {
```

Reply:
Sorry about that, I should have written this for the original commit!